pkg/asset-manager-utils/test/SinglePoolAaveATokenAssetManager.test.ts

```diff
@@ -0,0 +1,330 @@
+import { ethers } from 'hardhat';
+import { Contract } from 'ethers';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+
+import { bn, fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { MAX_UINT256 } from '@balancer-labs/v2-helpers/src/constants';
+
+import { deploy } from '@balancer-labs/v2-helpers/src/contract';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { expectBalanceChange } from '@balancer-labs/v2-helpers/src/test/tokenBalance';
+import { advanceTime } from '@balancer-labs/v2-helpers/src/time';
+import { encodeJoinWeightedPool } from '@balancer-labs/v2-helpers/src/models/pools/weighted/encoding';
+
+const OVER_INVESTMENT_REVERT_REASON = 'investment amount exceeds target';
+const UNDER_INVESTMENT_REVERT_REASON = 'withdrawal leaves insufficient balance invested';
+
+const tokenInitialBalance = bn(200e18);
+const amount = bn(100e18);
+
+const setup = async () => {
+  const [, admin, lp, other] = await ethers.getSigners();
```

Comment:
I'd have this function inlined inside the first `sharedBeforeEach` (and maybe split in smaller parts). That would also prevent mistakes caused by calling `getSigners` multiple times.